design/FY2022/enhacement_to_resilience_metric.md

```diff
@@ -0,0 +1,191 @@
+|...||||
+|Space<sub>N</sub>||||
+|Min||||
+|Max||||
+|Average||||
+|Sum||||
+<p style="text-align: center;"> Table 2. Sample report table of hours of safety for hot events.</p>
+
+**Unmet Degree Hours**
+
+|	Cooling Setpoint Unmet Degree Hours [°C•h] | Cooling Setpoint Unmet Degree OccupantHours [°C•h] | Heating Setpoint Unmet Degree Hours [°C•h] | Heating Setpoint Unmet Degree OccupantHours [°C•h]
+|-----------------------|---------------------------------|---------------------------------------|------------------|
+|Space<sub>1</sub>||||
+|...||||
+|Space<sub>N</sub>||||
+|Min||||
+|Max||||
+|Average||||
+|Sum||||
+<p style="text-align: center;"> Table 3. Sample report table of unmet degree hours (UDH).</p>
+
+**Discomfort-weighted Exceedance Hours**
+
+|	Very-cold Exceedance OccupantHours [h] | Cool Exceedance OccupantHours [h] | Warm Exceedance OccupantHours [h] | Very-hot Exceedance OccupantHours [h]
+|-----------------------|---------------------------------|---------------------------------------|------------------|
+|Space<sub>1</sub>||||
+|...||||
+|Space<sub>N</sub>||||
+|Min||||
+|Max||||
+|Average||||
+|Sum||||
+<p style="text-align: center;"> Table 4. Sample report table of discomfort-weighted exceedance hours.</p>
+
+When *ThermalResilienceSummary* is declared in *Output:Table:SummaryReports*, the above three tables will be generated and presented in the tabular reports. In particular, the *Hours of Safety* table will only be presented if an *Output:Table:SummaryReports:ReportPeriod* object is defined for ThermalResilienceSummary report. 
+
+## Testing/Validation/Data Source(s): ##
+
+Two example files (the DOE reference small office and the one zone uncontrolled model) will be modified to demonstrate the use of the new feature. Simulation results will be manually checked/benchmarked using an excel spreadsheet with input and output from EnergyPlus runs.
+
+## IDD Object changes: ##
+
+Two extra fields will be added to the current People object to define the safe temperature thresholds for a certain kind of population.
+
+    People,
+      N7, \field Cold Stress Temperature Threshold
+          \type real
+          \units C
+          \note this is the indoor safe temperature threshold for cold stress
+    \default 15.56
+      N8, \field Heat Stress Temperature Threshold
+          \type real
+          \units C
+          \note this is the indoor safe temperature threshold for heat stress
+          \default 30
+
+We will add a new object *Output:Table:SummaryReports:ReportPeriod* to add supplementary information to define the reporting period of certain summary tables.
+
+    Output:Table:SummaryReports:ReportPeriod,
+        \memo This object allows the user to add a reporting period to a certain tabular report when the reporting period is not the whole simulation period. When defined, the declared summary report is based on aggregations from the start date to the end date defined. 
+      A1, \field Report Name,
```

Comment:
If I'm following the intent correctly, a user may want to specify more than one reporting period. e.g. a summer power outage and a winter power outage. In that case, then field A1 should simply be Name (of the object), and Report Name would be A2. Another option would be to use a reporting schedule in this object instead of dates and times.  Either way, it will be important for the table outputs to include a note about the dates and number of hours the report was active.

Reply:
Good point. We'll have some internal discussion about how to go about enabling multiple reporting periods and update the feature design doc on this topic sometime next week. 

In the tabular report, under each table, a line of text, "Reporting period: xxx-xx to xxx-xx", is added to show what reporting period this table has. This line is a placeholder for now. It will be updated once we figure out the multiple report period design.

Reply:
The "Name" field is added to the idd object to allow for multiple reporting periods.

Reply:
I am concerned that OutputControl:Table:ReportPeriod is going to imply to people not assessing resilience at all that these are the report period for all of the output tables. I presume this is not the intention to modify all output table data collection periods based on this input, correct? If so, I think the name needs to be even more specific, how about: "OutputControl:Table:ReslienceReportPeriod"

Reply:
Hi @JasonGlazer, this ReportPeriod object is not specific to the resilience metric. Although currently we only allow for three keys (or maybe with the fourth one of AllResilienceMetric), we want it to be more generic and can be applied to other tables like energy reporting. For example, with this ReportPeriod feature, we can report energy usage by season.